xain_fl/coordinator/metrics_store.py

```diff
@@ -0,0 +1,127 @@
+    def write_metrics(self, participant_id: str, metrics: Dict[str, ndarray]) -> bool:
+        """Write the participant metrics on behalf of the participant with the given participant_id
+        into a metric store.
+
+        Args:
+
+            participant_id: The ID of the participant.
+            metrics: The metrics of the participant with the given participant_id.
+
+        Returns:
+
+            True, on success, otherwise False.
+        """
+
+
+class NullObjectMetricsStore(AbstractMetricsStore):  # pylint: disable=too-few-public-methods
+    """A metric store that does nothing."""
+
+    def write_metrics(self, participant_id: str, metrics: Dict[str, ndarray]) -> bool:
+        """A method that has no effect.
+
+        Args:
+
+            participant_id: The ID of the participant.
+            metrics: The metrics of the participant with the given participant_id.
+
+        Returns:
+
+            Always True.
+        """
+        return True
+
+
+class MetricsStore(AbstractMetricsStore):  # pylint: disable=too-few-public-methods
+    """A metric store that uses InfluxDB to store the metrics."""
+
+    def __init__(self, config: MetricsConfig):
+        self.config = config
+        self.influx_client = InfluxDBClient(
+            self.config.host,
+            self.config.port,
+            self.config.user,
+            self.config.password,
+            self.config.db_name,
+        )
+
+    def write_metrics(self, participant_id: str, metrics: Dict[str, ndarray]) -> bool:
+        """Write the participant metrics on behalf of the participant with the given participant_id
+        into InfluxDB.
+
+        Args:
+
+            participant_id: The ID of the participant.
+            metrics: The metrics of the participant with the given participant_id.
+
+        Returns:
+
+            True, on success, otherwise False.
+        """
+
+        # FIXME: We will change the data format of the metrics message in a separate ticket.
+        # The goal is, that coordinator doesn't need to transform the metrics anymore.
+        influx_data_points = transform_metrics_to_influx_data_points(participant_id, metrics)
+
+        try:
+            return self.influx_client.write_points(influx_data_points)
+        except Exception as err:  # pylint: disable=broad-except
+            logger.warn("Can not write metrics", error=str(err))
+            return False
+
+
+def transform_metrics_to_influx_data_points(
+    participant_id: str, metrics: Dict[str, ndarray]
+) -> List[dict]:
+    """Transform the metrics of a participant into InfluxDB data points.
+
+    Arguments:
+        participant_id: The ID of the participant.
+        metrics: The metrics of the participant with the given participant_id.
+
+    Returns:
+        The metrics of the participant as InfluxDB data points.
+    """
+
+    # Currently the metrics message does not contain any timestamps.
+    # We set a timestamp for each epoch data point with an interval of 1 sec.
```

Comment:
What is the reason for this?

Reply:
Since we are using a time series database, we need to include a timestamp for each data point. The interval thing is just for the case that the participant send metrics for more than 1 epoch per round. If we just use the same timestamp, all data points will group on a horizontal line.

Reply:
That makes sense, I think it is worth adding an explanation as a comment.

Reply:
I added it to the comment